fendermint/actors/chainmetadata/src/shared.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright 2021-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use cid::Cid;
+use fvm_ipld_encoding::tuple::{Deserialize_tuple, Serialize_tuple};
+use fvm_shared::{clock::ChainEpoch, METHOD_CONSTRUCTOR};
+use num_derive::FromPrimitive;
+
+// The state is a stores `blockhashes` in an AMT containing the blockhashes of the
+// last `lookback_len` epochs
+#[derive(Serialize_tuple, Deserialize_tuple)]
+pub struct State {
+    // the AMT root cid of blockhashes
+    pub blockhashes: Cid,
+
+    // the maximum size of blockhashes before removing the oldest epoch
+    pub lookback_len: u64,
+}
```

Comment:
Maybe this could also have a shared constructor like [this one](https://github.com/consensus-shipyard/ipc/blob/904705346b7788e79920526c95c0e62bee5869a3/fendermint/vm/actor_interface/src/init.rs#L53-L54) to take care of the empty array initialisation.